### PR TITLE
feat(runtimeconfig): support loading config from HTTP URLs

### PR DESCRIPTION
**Pull request by colega**
## Summary

The runtimeconfig Manager can now fetch config from HTTP/HTTPS URLs in addition to local filesystem paths. This enables centralized config distribution without requiring shared filesystems.

When a `LoadPath` entry starts with `http://` or `https://`, it is fetched via HTTP GET instead of `os.ReadFile`. A `provider` abstraction cleanly separates the two read mechanisms.

HTTP fetches are instrumented with a native histogram (`runtime_config_http_request_duration_seconds`) with `url` and `status_code` labels. On reload failure the previous config is retained; on first load failure the Manager fails to start.

New flag: `--runtime-config.http-client-timeout` (default 30s).

---

> [!NOTE]
> **Medium Risk**
> Adds network-based config loading with a new HTTP client, timeout setting, and request metrics; failures now depend on remote availability and HTTP semantics, which can affect startup/reload behavior.
> 
> **Overview**
> The runtime config `Manager` can now load and merge config from **either local files or `http(s)` URLs** in `LoadPath`, using a new `provider` abstraction (`fileProvider`/`httpProvider`).
> 
> This introduces a configurable HTTP client timeout (`--runtime-config.http-client-timeout`) and a new Prometheus histogram (`runtime_config_http_request_duration_seconds{url,status_code}`) to track fetch latency/outcomes, while updating hashing/error messages to key off provider names instead of file paths.
> 
> Tests are expanded to cover URL loading, mixed file+URL merges, first-load vs reload failure behavior (retaining prior config on reload failure), and HTTP provider edge cases (non-2xx, timeout, canceled context), plus some timing-related test stabilization via `synctest`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3fd169a74c5de4f78acaebdd589cabd11e8f616c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: runtimeconfig/manager_test.go [hunk 1]
```diff
@@ -742,3 +745,179 @@ func TestManager_GetConfigNilBeforeStarting(t *testing.T) {
...
+	t.Cleanup(func() { require.NoError(t, services.StopAndAwaitTerminated(context.Background(), manager)) })
+
+	require.Equal(t, value{Value: 42}, manager.GetConfig())
+
+	// Switch to failing.
+	mu.Lock()
+	statusCode = http.StatusInternalServerError
+	mu.Unlock()
+
+	time.Sleep(3 * reloadPeriod)
```
Comment (tcard) [inline runtimeconfig/manager_test.go]: It would be nice to use [synctest](https://pkg.go.dev/testing/synctest) for these.

**Review by tcard — APPROVED** [review]

File: runtimeconfig/manager_test.go (hunk 1 above)
Comment (colega) [reply runtimeconfig/manager_test.go]: Oh, I actually did this but didn't push the change.

**Comment by pracucci** [comment]
Hi! 👋 We recently merged #947 which changed the CI configuration for unit tests. To get CI passing on this PR, please rebase on the latest `main` branch:

```
git fetch origin main
git rebase origin/main
git push --force-with-lease
```

Thanks!
